.github/workflows/clojure-linting.yaml

```diff
@@ -0,0 +1,33 @@
+name: Clojure Linting
+
+on:
+  push:
+    branches: [main]
```

Comment:
I think that running this again only on `main` when merged/force-pushed is a bit unnecessary and gets in to the mess of trying to keep the branches straight here (we are already behind on https://github.com/puppetlabs/orchestrator/blob/e830895efecd06b4eb24fb9b376a8fd37b610d34/.github/workflows/clojure-linting.yaml#L5 ). Specifically i think we decided you have to list all the branches you want in all the files across branches that define a workflow with a common name. 

I think it makes sense to only run this on pull request, if you want to run in every time code lands in a branch we will have to include the list of all active branches under `on.push.branches` i think.

Reply:
Good point. And I missed adding 7.x to it! I'll update